mvcc/kvstore_txn.go

```diff
@@ -133,6 +157,15 @@ func (tr *storeTxnRead) rangeKeys(key, end []byte, curRev int64, ro RangeOptions
 		limit = len(revpairs)
 	}
 
+	if limit > expensiveReadLimit && !tr.txlocked && tr.ro { // first expensive read in a read only transcation
```

Comment:
e.g., `storeTxnReadLimited` could impose a limit+1 on `kvindex.Revisions()`. If the result is >limit, load by chunk from the committed tx-- it would split up the rlock hold times and spend less memory on `revpairs`